### PR TITLE
CNV-14984: verbose logs with the CLI

### PR DESCRIPTION
**Pull request by ousleyp**
Version(s): 4.11+

Issue: https://issues.redhat.com/browse/CNV-14984

Link to docs preview: http://file.rdu.redhat.com/pousley/cnv-14984/virt/logging_events_monitoring/virt-logs.html

Note: The troubleshooting module is out of scope for this story, but I created it as a stub/WIP so that the error info would have a more logical home. (Most of the module text was previously a warning in the assembly.)

### CODE REVIEW

File: modules/virt-viewing-component-logs-cli.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  configuration:
+    developerConfiguration:
+      logVerbosity:
+        virtLauncher: 2 <1>
+        virtHandler: 3
+        virtController: 4
+        virtAPI: 5
+        virtOperator: 6
+----
+<1> The log verbosity value must be an integer in the range `1–9`, where a higher number indicates a more detailed log. In this example, the `virtLauncher` component logs are exposed if they are greater than or equal to a verbosity level of `2`.
```
Comment (iholder-redhat) [inline modules/virt-viewing-component-logs-cli.adoc:38]: I just want to point out that, at least in my opinion, we should determine a much clearer definition of each verbosity level. I've also opened [an issue](https://github.com/kubevirt/kubevirt/issues/8164) on that subject.

As I see it, we would end up with something like 4-5 level that are `Info`, `Debug`, `Warnings`, etc.

Maybe that's worth mentioning here? @stu-gott WDYT?

**Review by iholder101 — COMMENTED** [review]
Looks great
/lgtm

File: modules/virt-viewing-component-logs-cli.adoc (hunk 1 above)
Comment (ousleyp) [reply modules/virt-viewing-component-logs-cli.adoc:38]: +1, I agree that the levels should be more clearly defined. We shouldn't mention features that aren't implemented yet, but we'd be happy to update the docs later.

File: modules/virt-viewing-logs-cli.adoc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  name: kubevirt-hyperconverged
+spec:
+  configuration:
+    developerConfiguration:
+      logVerbosity:
+        virtLauncher: 2 <1>
+        virtHandler: 3
+        virtController: 4
+        virtAPI: 5
+        virtOperator: 6
```
Comment (dshchedr) [inline modules/virt-viewing-logs-cli.adoc:36]: This yaml exemple is incorrect, the correct one:

```
spec
  logVerbosityConfig:
    kubevirt:
      virtAPI: 6
      virtController: 6
      .
      .
```

File: modules/virt-viewing-logs-cli.adoc [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+----
+<1> The log verbosity value must be an integer in the range `1–9`, where a higher number indicates a more detailed log. In this example, the `virtLauncher` component logs are exposed if they are greater than or equal to a verbosity level of `2`.
```
[the remark this answers is not in the document]
Comment (ousleyp) [reply modules/virt-viewing-logs-cli.adoc:38]: Thanks, Stu! Did I interpret this correctly? I made the following changes (including the fixed YAML):

```
[source,yaml]
----
apiVersion: hco.kubevirt.io/v1beta1
kind: HyperConverged
metadata:
  name: kubevirt-hyperconverged
spec:
  logVerbosityConfig:
    kubevirt:
      virtAPI: 5 <1>
      virtController: 4
      virtHandler: 3
      virtLauncher: 2
      virtOperator: 6
----
<1> The log verbosity value must be an integer in the range `1–9`, where a higher number indicates a more detailed log. In this example, the `virtAPI` component logs are exposed if their priority level is `5` or higher.
```

**Comment by dshchedr** [comment]
looks great, thanks
/lgtm

File: modules/virt-viewing-logs-cli.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (stu-gott) [reply modules/virt-viewing-logs-cli.adoc]: Yes this works. Thanks!

File: modules/virt-viewing-logs-cli.adoc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+[NOTE]
+====
+If a pod fails to start, you can use the `--previous` option to view logs from the last attempt.
+
+To monitor log output in real time, use the `-f` option.
+====
++
+.Example output
+[source,terminal]
+[%collapsible]
```
Comment (sjhala-ccs) [inline modules/virt-viewing-logs-cli.adoc:87]: If you are planning to use the `collapsible` option, the code block should be between ==== formatting as per our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#collapsible-content)

**Review by sjhala-ccs — COMMENTED** [review]
Just a couple of nits, otherwise LGTM

File: modules/virt-viewing-logs-cli.adoc (hunk 4 above)
Comment (ousleyp) [reply modules/virt-viewing-logs-cli.adoc:87]: Thanks for the heads up! I wasn't aware of this.

**Pull request merged**